enf/enf.go

```diff
@@ -69,12 +70,17 @@ type service struct {
 // For usage examples, see the methods in network.go or firewall.go
 
 // get makes a get request to the given path and stores the response in the given body object.
-func (c *Client) get(ctx context.Context, path string, body interface{}) (interface{}, *http.Response, error) {
+func (c *Client) get(ctx context.Context, path string, body interface{}, queryParameters url.Values) (interface{}, *http.Response, error) {
```

Comment:
I'm ok with this approach, but consider these three other options as well.

1) A separate function `getWithQuery(ctx, path, query, body)`.  Then the users of the plain `get(ctx, path, body)` variant aren't complicated with an extra `nil` argument.
1) Reorder the arguments to `get(ctx, path, query, body)`.  `path` and `query` together form the URL, so it seems more natural to have them consecutive and before `body` in the parameter list.
1) Require an empty `url.Values` map instead of `nil`.  Callers would look like `get(ctx, path, url.Values{}, inviteResponse)`, instead of `get(ctx, path, nil, inviteResponse)`. I find this easier to read, since the empty query argument becomes self-documenting. The `nil` doesn't say anything about what could go there.  The implementation could then drop the `nil` check for `queryParameters`.

There's nothing wrong with the way you've done it.  And in my limited knowledge, it seems like fairly idiomatic go code.  Just some other options to consider, if you haven't already.

Reply:
> In the future, this should be two commits in the PR.

I agree with this. In retrospect, it definitely should have been two commits. I'll keep that in mind for the future. 

I really like your suggestions for the refactor. Here's what I've settled on. 

1. After thinking about it, I'm going to keep the single function to be consistent across the codebase.
2. I'll reorder the arguments to make the function call more intuitive. This is a subtle but effective change.
3. I'll have the `get` method take in an empty `url.Values` instead of `nil`. You're right in that we wouldn't have to do the check on our side and instead just use whatever value the caller provides for the query parameters.

Once again, great suggestions!